Mathlib/Algebra/Group/Units.lean

```diff
@@ -727,6 +727,15 @@ theorem isUnit_iff_exists_inv' [CommMonoid M] {a : M} : IsUnit a ↔ ∃ b, b *
 #align is_unit_iff_exists_inv' isUnit_iff_exists_inv'
 #align is_add_unit_iff_exists_neg' isAddUnit_iff_exists_neg'
 
+@[to_additive]
+theorem isUnit_iff_exists_and_exists [Monoid M] {a : M} :
+    IsUnit a ↔ (∃ b, a * b = 1) ∧ (∃ c, c * a = 1):= by
+  constructor
+  · rintro ⟨⟨a, b, hab, hba⟩, rfl⟩
+    exact ⟨⟨b, hab⟩, ⟨b, hba⟩⟩
+  · rintro ⟨⟨b, hab⟩, ⟨c, hca⟩⟩
+    exact ⟨⟨a, b, hab, left_inv_eq_right_inv hca hab ▸ hca⟩, rfl⟩
```

Comment:
```suggestion
  refine ⟨fun h => ⟨h.exists_right_inv, h.exists_left_inv⟩, ?_⟩
  rintro ⟨⟨b, hab⟩, ⟨c, hca⟩⟩
  exact ⟨⟨a, b, hab, left_inv_eq_right_inv hca hab ▸ hca⟩, rfl⟩
```
I propose one further golf, since some of these deconstruction/construction lemmas already exist!